src/sdk/namespace_fs.js

```diff
@@ -309,9 +360,25 @@ const versions_dir_cache = new LRUCache({
         if (stat.size + ver_dir_stat_size <= config.NSFS_DIR_CACHE_MAX_DIR_SIZE) {
             const latest_versions = await nb_native().fs.readdir(fs_context, dir_path);
             if (is_version_path_exists) {
-                const old_versions = await nb_native().fs.readdir(fs_context, version_path);
-                const entries = latest_versions.concat(old_versions);
+                let old_versions = await nb_native().fs.readdir(fs_context, version_path);
+                // In case we have a null version id inside .versions/ directory we will rename it
+                // Therefore, old_versions will not include an entry with 'null' suffix
+                const { renamed_old_null_versions, error_on_stat } = await _rename_null_version(old_versions, fs_context, version_path);
+                // Remove null versions that could not be rename (error on stat) and their name changed to undefined
+                if (error_on_stat) old_versions = old_versions.filter(old_version => old_version.name);
```

Comment:
hmm so that's another iteration on a possibly very large array, can you maybe use reduce instead?

Reply:
Why would we prefer `reduce` over `filter` in this case?
It would be less readable: 
```suggestion
                if (error_on_stat) old_versions = old_versions.reduce((accumulator, old_version) => {
                    if (old_version.name) accumulator.push(old_version);
                    return accumulator;
                  }, []);
```

Reply:
I meant that in order to avoid another iteration on the array you can do the filter inside _rename_null_version in a single reduce function.

Reply:
Thanks, I change it so we won't do another iteration in case stat fails.